ObjectivePGP/NSData+compression.m

```diff
@@ -158,26 +163,33 @@ - (NSData *)bzip2Compressed:(NSError * __autoreleasing *)error
     stream.next_out = [buffer mutableBytes];
     stream.avail_out = buffer_size;
     
-    NSMutableData * decompressed = [NSMutableData data];
+    bzret = BZ2_bzCompressInit(&stream, compression, 0, 0);
+    if (bzret != BZ_OK) {
+        if (error) {
+            *error = [NSError errorWithDomain:@"BZIP2" code:bzret userInfo:@{ NSLocalizedDescriptionKey: NSLocalizedString(@"BZ2_bzCompressInit failed", nil) }];
+        }
+        return nil;
+    }
+    
+    NSMutableData *compressedData = [NSMutableData data];
     
-    BZ2_bzCompressInit(&stream, compression, 0, 0);
     do {
         bzret = BZ2_bzCompress(&stream, (stream.avail_in) ? BZ_RUN : BZ_FINISH);
-        if (bzret != BZ_RUN_OK && bzret != BZ_STREAM_END) {
+        if (bzret < BZ_OK) {
```

Comment:
Yes, that is the key change. If the compressed file exceeds 10000 bytes, the return code becomes BZ_FLUSH_OK and the original code marks that as an error, while it is not.

Less important is that this code also checks for errors on the CompressInit.

Feel free to remove the: NSLocalizedDescriptionKey, I should have taken that out myself.